underfs/oss/src/main/java/alluxio/underfs/oss/OSSActiveSyncProvider.java

```diff
@@ -0,0 +1,356 @@
+        ObjectListing objectListing = mClient.listObjects(bucketName, prefix);
+        List<OSSObjectSummary> sums = objectListing.getObjectSummaries();
+        for (OSSObjectSummary s : sums) {
+          if (isChanged(ufsUri, s.getLastModified().getTime())) {
+            updateUriLastModified(ufsUri, s.getLastModified().getTime());
+            filesUri.add(new AlluxioURI(s.getKey()));
+            LOG.info("\t" + s.getKey() + " is changed. ");
+          }
+        }
+        return filesUri;
+      } else {
+        List<AlluxioURI> filesUri = new ArrayList<>();
+        String bucketName = UnderFileSystemUtils.getBucketName(ufsUri);
+        String fileName = cutSlash(ufsUri.getPath());
+        if (mClient.doesObjectExist(bucketName, fileName)) {
+          OSSObject ossObject = mClient.getObject(bucketName, ufsUri.getPath());
+          if (null != ossObject
+              && isChanged(ufsUri, ossObject.getObjectMetadata().getLastModified().getTime())) {
+            updateUriLastModified(ufsUri,
+                ossObject.getObjectMetadata().getLastModified().getTime());
+            filesUri.add(new AlluxioURI(ossObject.getKey()));
+            LOG.info("\t" + ossObject.getKey() + " is changed. ");
+          }
+        }
+        return filesUri;
+      }
+    } catch (OSSException oe) {
+      LOG.info("Caught an OSSException, which means your request made it to OSS, "
+          + "but was rejected with an error response for some reason.");
+      LOG.info("Error Message:" + oe.getErrorMessage());
+      LOG.info("Error Code:" + oe.getErrorCode());
+      LOG.info("Request ID:" + oe.getRequestId());
+      LOG.info("Host ID:" + oe.getHostId());
+      return Collections.emptyList();
+    } catch (ClientException ce) {
+      LOG.info("Caught an ClientException, which means the client encountered "
+          + "a serious internal problem while trying to communicate with OSS, "
+          + "such as not being able to access the network.");
+      LOG.info("Error Message:" + ce.getMessage());
+      return Collections.emptyList();
+    }
+  }
+
+  private boolean isDirectory(AlluxioURI ufsUri) {
+    String bucketName = UnderFileSystemUtils.getBucketName(ufsUri);
+    String fileName = ufsUri.getPath();
+    fileName = cutSlash(fileName);
+    if (mClient.doesObjectExist(bucketName, fileName + mOssFolderSuffix)) {
+      return true;
+    }
+    return false;
+  }
+
+  private String cutSlash(String fileName) {
+    String newFileName = fileName;
+    if (newFileName.startsWith("/")) {
+      newFileName = newFileName.substring(1);
+    }
+    if (newFileName.endsWith("/")) {
+      newFileName = newFileName.substring(0, newFileName.length() - 1);
+    }
+    return newFileName;
+  }
+
+  private boolean isChanged(AlluxioURI ufsUri, long ufsLastModifiedTimestamp) {
+    Long lastModified = mUfsUriLastModified.get(ufsUri);
+    if (null == lastModified || lastModified != ufsLastModifiedTimestamp) {
```

Comment:
I think it's better to get the latest time between them. so it would be like 
`if (null == lastModified || lastModified < ufsLastModifiedTimestamp) return true`

Reply:
Yes, it is. It would be better to use `<` instead of `!=`.